<!DOCTYPE html>
<html lang="zh-cn" itemscope itemtype="http://schema.org/WebPage">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>病原微生物扩增子数据分析实战（三）：vsearch软件鉴定物种组成 - 简佐义的博客</title>
  

<meta name="renderer" content="webkit" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>

<meta name="MobileOptimized" content="width"/>
<meta name="HandheldFriendly" content="true"/>


<meta name="applicable-device" content="pc,mobile">

<meta name="theme-color" content="#f8f5ec" />
<meta name="msapplication-navbutton-color" content="#f8f5ec">
<meta name="apple-mobile-web-app-capable" content="yes">
<meta name="apple-mobile-web-app-status-bar-style" content="#f8f5ec">

<meta name="mobile-web-app-capable" content="yes">

<meta name="author" content="简佐义" />
  <meta name="description" content="病原微生物基因检测的两大核心任务是物种组成和功能组成的鉴定，而扩增子测序的首要目的是找到致病的细菌或者病毒，即鉴定物种组成。 经过质量控制，测" />

  <meta name="keywords" content="生物信息学, 自学生物信息学, 生物信息学学习, 生物学, linux, python, R, 简佐义, 博客, 笔记" />






<meta name="generator" content="Hugo 0.79.0" />


<link rel="canonical" href="https://jianzuoyi.github.io/post/2020-09-14.microbe-vsearch/" />





<link rel="icon" href="/favicon.ico" />











<link rel="stylesheet" href="/sass/jane.min.b3a8813c06e6d785beba22bf8264e174fa2cb3a396b22f9ba24e2c00c18aaf7f.css" integrity="sha256-s6iBPAbm14W&#43;uiK/gmThdPoss6OWsi&#43;bok4sAMGKr38=" media="screen" crossorigin="anonymous">





<meta property="og:title" content="病原微生物扩增子数据分析实战（三）：vsearch软件鉴定物种组成" />
<meta property="og:description" content="病原微生物基因检测的两大核心任务是物种组成和功能组成的鉴定，而扩增子测序的首要目的是找到致病的细菌或者病毒，即鉴定物种组成。 经过质量控制，测" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://jianzuoyi.github.io/post/2020-09-14.microbe-vsearch/" />
<meta property="article:published_time" content="2020-09-14T00:00:00+08:00" />
<meta property="article:modified_time" content="2020-12-15T20:23:54+08:00" />
<meta itemprop="name" content="病原微生物扩增子数据分析实战（三）：vsearch软件鉴定物种组成">
<meta itemprop="description" content="病原微生物基因检测的两大核心任务是物种组成和功能组成的鉴定，而扩增子测序的首要目的是找到致病的细菌或者病毒，即鉴定物种组成。 经过质量控制，测">
<meta itemprop="datePublished" content="2020-09-14T00:00:00+08:00" />
<meta itemprop="dateModified" content="2020-12-15T20:23:54+08:00" />
<meta itemprop="wordCount" content="2789">



<meta itemprop="keywords" content="病原微生物检测,自学生信," />
<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="病原微生物扩增子数据分析实战（三）：vsearch软件鉴定物种组成"/>
<meta name="twitter:description" content="病原微生物基因检测的两大核心任务是物种组成和功能组成的鉴定，而扩增子测序的首要目的是找到致病的细菌或者病毒，即鉴定物种组成。 经过质量控制，测"/>

<!--[if lte IE 9]>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/classlist/1.1.20170427/classList.min.js"></script>
<![endif]-->

<!--[if lt IE 9]>
  <script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/respond.js@1.4.2/dest/respond.min.js"></script>
<![endif]-->




</head>
<body>
  <div id="mobile-navbar" class="mobile-navbar">
  <div class="mobile-header-logo">
    <a href="/" class="logo">简佐义的博客</a>
  </div>
  <div class="mobile-navbar-icon">
    <span></span>
    <span></span>
    <span></span>
  </div>
</div>
<nav id="mobile-menu" class="mobile-menu slideout-menu">
  <ul class="mobile-menu-list">
    <li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/">首页</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/categories/">分类</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/tags/">标签</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/post/">归档</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/about/">关于</a>
          
        
      </li><li class="mobile-menu-item">
        
          
          <div class="mobile-menu-parent">
            <span class="mobile-submenu-open"></span>
            <a href="https://jianzuoyi.github.io/">
              精选&#43;
            </a>
          </div>
          <ul class="mobile-submenu-list">
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-23-learn-bioinfomatics/">01 如何自学生物信息学：从菜鸟到专家</a>
              </li>
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-20.linux/">02 生信人的自我修养：Linux 命令速查手册</a>
              </li>
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-19-conda/">03 生物信息基础：conda包管理器安装及使用</a>
              </li>
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-09.git/">04 代码版本控制及托管：我的最佳实践</a>
              </li>
            
          </ul>
        
      </li>
    

    
  </ul>
</nav>


  
    






  <link rel="stylesheet" href="/lib/photoswipe/photoswipe.min.css" />
  <link rel="stylesheet" href="/lib/photoswipe/default-skin/default-skin.min.css" />




<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>

  

  

  
    <style>
      .fork-me-on-github {
        position: absolute; top: 0; left: 0; border: 0;
      }
      @media (max-width: 1080px) {
        .fork-me-on-github {
          display: none;
        }
      }
    </style>

    <a href="https://github.com/jianzuoyi">
      <img class="fork-me-on-github"
        src="https://s3.amazonaws.com/github/ribbons/forkme_left_red_aa0000.png"
        alt="Fork me on GitHub">
    </a>
  

  <header id="header" class="header container">
    <div class="logo-wrapper">
  <a href="/" class="logo">
    
      简佐义的博客
    
  </a>
</div>

<nav class="site-navbar">
  <ul id="menu" class="menu">
    
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/">首页</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/categories/">分类</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/tags/">标签</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/post/">归档</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          
            <a class="menu-item-link" href="https://jianzuoyi.github.io/about/">关于</a>
          

        

      </li>
    
        <li class="menu-item">
        
          
          <a class="menu-item-link menu-parent" href="https://jianzuoyi.github.io/">精选&#43;</a>
          <ul class="submenu">
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-23-learn-bioinfomatics/">01 如何自学生物信息学：从菜鸟到专家</a>
              </li>
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-20.linux/">02 生信人的自我修养：Linux 命令速查手册</a>
              </li>
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-19-conda/">03 生物信息基础：conda包管理器安装及使用</a>
              </li>
            
              <li>
                <a href="https://jianzuoyi.github.io/post/best/2020-11-09.git/">04 代码版本控制及托管：我的最佳实践</a>
              </li>
            
          </ul>

        

      </li>
    

    
    

    
  </ul>
</nav>

  </header>

  <div id="mobile-panel">
    <main id="main" class="main bg-llight">
      <div class="content-wrapper">
        <div id="content" class="content container">
          <article class="post bg-white">
    
    <header class="post-header">
      <h1 class="post-title">病原微生物扩增子数据分析实战（三）：vsearch软件鉴定物种组成</h1>
      
      <div class="post-meta">
        <time datetime="2020-09-14" class="post-time">
          2020-09-14
        </time>
        <div class="post-category">
            <a href="https://jianzuoyi.github.io/categories/%E7%97%85%E5%8E%9F%E5%BE%AE%E7%94%9F%E7%89%A9%E6%A3%80%E6%B5%8B/"> 病原微生物检测 </a>
            
          </div>
        

        
        
          <span id="busuanzi_container_page_pv">
            | 阅读 <span id="busuanzi_value_page_pv"></span>
          </span>
        

        
        
      </div>
    </header>

    
    
<div class="post-toc" id="post-toc">
  <h2 class="post-toc-title">文章目录</h2>
  <div class="post-toc-content">
    <nav id="TableOfContents">
  <ul>
    <li>
      <ul>
        <li></li>
      </ul>
    </li>
  </ul>
</nav>
  </div>
</div>

    
    <div class="post-content">
      <p>病原微生物基因检测的两大核心任务是物种组成和功能组成的鉴定，而扩增子测序的首要目的是找到致病的细菌或者病毒，即鉴定物种组成。</p>
<p>经过质量控制，测序数据中已经不包含非生物的异常序列了，下面我们用<a href="https://github.com/torognes/vsearch">vsearch</a>软件完全后续分析。</p>
<h4 id="1去重dereplication">1.去重（dereplication）</h4>
<p>同一对引物的扩增产物，理论上应该是完全一样的，这些冗余的信息会造成比较大的运算负担，因此需要先去冗余，相同的序列只保留一条就好了。</p>
<p>vsearch去冗余有以下三种模式：</p>
<ul>
<li>
<p>全长模式：两条序列长度完全相同，碱基完全一样，总之，序列要完全一模一样才认为是重复</p>
</li>
<li>
<p>一致模式：在全长模式的基础上，进一步要求序列的名称要完全相同</p>
</li>
<li>
<p>前缀模式：只要一条序列是另一条序列的前缀，也就是一条序列与另一条序列的前面部分相同，就可以认为它们是重复。</p>
</li>
</ul>
<p>可以看出，这三种模式的严格程度依次是：一致模式 &gt; 全长模式 &gt; 前缀模式。</p>
<p>我们用全长模式去除冗余：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">vsearch --derep_fulllength <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_merge.fastq.gz --output <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_derep.fa --minuniquesize <span class="m">8</span> --minseqlength <span class="m">100</span> --strand both --sizeout --fasta_width <span class="m">0</span> 
</code></pre></td></tr></table>
</div>
</div><p><code>--derep_fulllength</code>，后面跟通过质控的fastq文件；</p>
<p><code>--output</code>，输出文件，fasta格式；</p>
<p><code>--minuniquesize</code>，最低丰度值，低于该丰度的序列会被过滤掉；</p>
<p><code>--minseqlength</code>，最低长度值，低于该长度的序列会被过滤掉；</p>
<p><code>--strand</code>，当判断两条序列是否一致时，默认只考虑正链<code>plus</code>，<code>both</code>表示考虑正反两个方向都考虑；</p>
<p><code>--sizeout</code>，在结果文件中序列名称后面添加丰度信息；</p>
<p><code>--fasta_width</code>，限定fasta结果文件中每条序列在一行中最多显示的字符数，默认是80，0表示不做限制；</p>
<h4 id="2降噪denoise">2.降噪（denoise）</h4>
<p>按97%相似度对序列进行聚类曾经是扩增子序列分析的金标准，但这有一个问题，就是物种只能鉴定到属或种，因为有些物种之间的相似度接近99%，甚至有的菌株之间只有SNP的差别。因此现在流行的做法是进行降噪处理，通俗地说就是按100%相似度进行聚类，从而使鉴定达到种或株的水平。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">vsearch --cluster_unoise <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_derep.fa --centroids <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_cent.fa --consout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_cons.fa --minsize <span class="m">8</span> --strand both --sizeout --fasta_width <span class="m">0</span> --clusterout_sort
</code></pre></td></tr></table>
</div>
</div><p><code>--cluster_unoise</code>，上一步去重后的fasta文件；</p>
<p><code>--centroids</code>，fasta结果文件，包含每一个聚类中的种子序列；</p>
<p><code>--consout</code>，fasta结果文件，包含每一个聚类的一致性序列；</p>
<p><code>--minsize</code>，降噪最低的序列丰度要求，默认是8条；</p>
<p><code>--strand</code>，当判断两条序列是否一致时，默认只考虑正链<code>plus</code>，<code>both</code>表示考虑正反两个方向都考虑；</p>
<p><code>--sizeout</code>，在结果文件中序列名称后面添加丰度信息；</p>
<p><code>--fasta_width</code>，限定fasta结果文件中每条序列在一行中最多显示的字符数，默认是80，0表示不做限制；</p>
<p><code>--clusterout_sort</code>，结果文件中序列的顺序默认是按其在输入文件中的顺序，设定该参数则是按照降噪后序列的丰度排序；</p>
<h4 id="3去嵌合体">3.去嵌合体</h4>
<p>测序数据中可能存在嵌合体序列，每个嵌合体至少来源于两个或两个以上的扩增子模板，因而需要事先去除。</p>
<p>去嵌合体有两种方法：一是基于参考数据库，一是基于无参考数据库的denovo方法。</p>
<p>vsearch基于denovo方法去除嵌合体的命令：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">vsearch --uchime3_denovo <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_cons.fa --nonchimeras <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_nochimeras.fa --chimeras <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_chimeras.fa --uchimealns <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_chimeras.aln --sizeout --fasta_width <span class="m">0</span>
</code></pre></td></tr></table>
</div>
</div><p><code>--uchime3_denovo</code>，上一步去重后获得的一致性序列文件；</p>
<p><code>--nonchimeras</code>，不含有嵌合体的结果文件；</p>
<p><code>--chimeras</code>，含有嵌合体的结果文件；</p>
<p><code>--uchimealns</code>，以人类易于阅读的形式呈现嵌合体与其两个亲本进行比对的结果文件；</p>
<p><code>--sizeout</code>，在结果文件中序列名称后面添加丰度信息；</p>
<p><code>--fasta_width</code>，限定fasta结果文件中每条序列在一行中最多显示的字符数，默认是80，0表示不做限制；</p>
<h4 id="4创建otu表">4.创建OTU表</h4>
<p>OTU（operational taxonomic units），即可操作分类单元，通过vsearch创建OTU表，可以看到每个OTU的丰度情况。</p>
<p>a.拷贝OTU文件：经过去重、降噪、去嵌合体之后，获得的无嵌合体的fasta文件可作为最终OTU结果</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">cp <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_nochimeras.fa <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_otu.fa
</code></pre></td></tr></table>
</div>
</div><p>b.格式转换：将fastq格式的质控文件转换成fasta格式，以下命令只做格式转换，没有进行任何过滤操作</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">vsearch --fastx_filter <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_merge_clean_final.fastq.gz --fastaout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_merge_clean_final.fa
</code></pre></td></tr></table>
</div>
</div><p>c.创建OTU表：本质上是将经过质控的序列比对到OTU序列上，从而可以得出每一个OTU序列的丰度</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">vsearch --usearch_global <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_merge_clean.fa --db <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_otu.fa --id 0.97 --query_cov 0.97 --strand both --otutabout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_otutab.txt --samheader --samout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.sam
</code></pre></td></tr></table>
</div>
</div><p><code>--usearch_global</code>，后面跟经过质控的fastq文件，fasta格式；</p>
<p><code>--db</code>，OTU文件，fasta格式；</p>
<p><code>--id</code>，相似度阈值：当查询序列与目标序列之间的相似度达到多少时，才算比对上；</p>
<p><code>--query_cov</code>，覆盖度：满足相似度的情况下，同时要求查询序列的覆盖度达到多少；</p>
<p><code>--strand</code>，当判断两条序列是否一致时，默认只考虑正链<code>plus</code>，<code>both</code>表示考虑正反两个方向都考虑；</p>
<p><code>--outtabout</code>，输出OTU表。</p>
<p><code>--samheader</code>，当要保存SAM文件时，默认不包含SAM文件头，设定此参数即会包含；</p>
<p><code>--samout</code>，输出比对的SAM格式结果文件，可用于后续IGV软件可视化查看比对情况。</p>
<h4 id="5物种鉴定">5.物种鉴定</h4>
<p>有了样本的代表性序列后，想知道样本的物种组成，可将这些序列比对到参考数据库，通过相似性及覆盖度进行判断。</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">vsearch --usearch_global <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_otu.fa -db <span class="nv">$DB</span>/amplicon.fa --id 0.97 --query_cov 0.97 --strand both --biomout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_otu_tax.txt --alnout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>.aln --blast6out <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_blast.xls --fastapairs <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_pairs.fa --notmatched <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_notmatched.fa --userfields query+target+id+qcov+tcov --userout <span class="si">${</span><span class="nv">outdir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_stat.xls
</code></pre></td></tr></table>
</div>
</div><p><code>--usearch_global</code>，OTU文件，fasta格式；</p>
<p><code>--db</code>，参考序列库，fasta格式；</p>
<p><code>--id</code>，相似度阈值：当查询序列与目标序列之间的相似度达到多少时，才算比对上；</p>
<p><code>--query_cov</code>，覆盖度：满足相似度的情况下，同时要求查询序列的覆盖度达到多少；</p>
<p><code>--strand</code>，当判断两条序列是否一致时，默认只考虑正链<code>plus</code>，<code>both</code>表示考虑正反两个方向都考虑；</p>
<p><code>--biomout</code> ，输出biom格式的结果文件；</p>
<p><code>--alnout</code>，输出比对结果文件；</p>
<p><code>--blast6out</code>，输出blast 6格式的比对结果文件；</p>
<p><code>--fastapairs</code>，fasta格式文件，保存了查询序列以及对应的目标序列；</p>
<p><code>--notmatched</code>，没有搜索到目标序列的查询序列集合；</p>
<p><code>--userfields</code>，自定义输出结果的列名，从左至右分别为：查询序列id，目标序列id，相似度，查询序列覆盖度，目标序列覆盖度；</p>
<p><code>--userout</code>，按&ndash;userfields定义的表头输出自定义的结果文件。</p>
<p>上述命令得到一个自定义的结果文件<code>${outdir}/${sample}_stat.xls</code>，根据其中的相似度、覆盖度以及前面的OTU表<code>${outdir}/${sample}_otutab.txt</code>，就可以知到物种的组成情况，以及它们的丰度信息。</p>
<p>现在，通过3篇文章，一套病原微生物扩增子物种鉴定的流程就建立起来了。</p>
<p><strong>往期精彩</strong></p>
<p><a href="https://mp.weixin.qq.com/s?__biz=MzIxMjQxMDYxNA==&amp;mid=2247483897&amp;idx=1&amp;sn=d4db2ad66feedbbea6587f0023d5c4a0&amp;chksm=9747c860a0304176ca0197a17ad0fb3c523ecba781ef56248eb66080cfcf811f1586197b7b24&amp;token=123359481&amp;lang=zh_CN#rd">病原微生物扩增子数据分析实战（二）：fastp软件进行质量控制</a></p>
<p><a href="https://mp.weixin.qq.com/s?__biz=MzIxMjQxMDYxNA==&amp;mid=2247483860&amp;idx=1&amp;sn=75f5503b79466b32d0897f6b0e831f42&amp;chksm=9747c84da030415b822c3b086304f0e61a7e41bd34bfc903e9e32a933227004d958433a35f56&amp;token=123359481&amp;lang=zh_CN#rd">病原微生物扩增子数据分析实战（一）：bcl2fastq软件完成数据拆分</a></p>

    </div>

    
    
<div class="post-copyright">
  <p class="copyright-item">
    <span class="item-title">文章作者</span>
    <span class="item-content">简佐义</span>
  </p>
  <p class="copyright-item">
    <span class="item-title">上次更新</span>
    <span class="item-content">
      2020-12-15
      
        <a href="https://github.com/jianzuoyi/jianzuoyi.github.io/commit/de676deb56ad610b001429cdec1dc2865f38d284" title="chmod">
          (de676de)
        </a>
        
          <br>chmod
        
      
    </span>
  </p>
  
  <p class="copyright-item">
    <span class="item-title">许可协议</span>
    <span class="item-content">本站文章皆为原创，转载需注明作者及文章链接，谢谢！</span>
  </p>
</div>


    
    

    <footer class="post-footer">
      <div class="post-tags">
          <a href="https://jianzuoyi.github.io/tags/%E7%97%85%E5%8E%9F%E5%BE%AE%E7%94%9F%E7%89%A9%E6%A3%80%E6%B5%8B/">病原微生物检测</a>
          <a href="https://jianzuoyi.github.io/tags/%E8%87%AA%E5%AD%A6%E7%94%9F%E4%BF%A1/">自学生信</a>
          
        </div>

      
      <nav class="post-nav">
        
          <a class="prev" href="/post/2020-09-15.aspera/">
            
            <i class="iconfont">
              <svg  class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M691.908486 949.511495l75.369571-89.491197c10.963703-12.998035 10.285251-32.864502-1.499144-44.378743L479.499795 515.267417 757.434875 204.940602c11.338233-12.190647 11.035334-32.285311-0.638543-44.850487l-80.46666-86.564541c-11.680017-12.583596-30.356378-12.893658-41.662889-0.716314L257.233596 494.235404c-11.332093 12.183484-11.041474 32.266891 0.657986 44.844348l80.46666 86.564541c1.772366 1.910513 3.706415 3.533476 5.750981 4.877077l306.620399 321.703933C662.505829 963.726242 680.945807 962.528973 691.908486 949.511495z"></path>
</svg>

            </i>
            <span class="prev-text nav-default">Aspera：基因组数据高速下载利器，以NCBI和EBI数据下载为例</span>
            <span class="prev-text nav-mobile">上一篇</span>
          </a>
        
          <a class="next" href="/post/2020-09-05.microbe-fastp/">
            <span class="next-text nav-default">病原微生物扩增子数据分析实战（二）：fastp软件进行质量控制</span>
            <span class="prev-text nav-mobile">下一篇</span>
            
            <i class="iconfont">
              <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="18" height="18">
  <path d="M332.091514 74.487481l-75.369571 89.491197c-10.963703 12.998035-10.285251 32.864502 1.499144 44.378743l286.278095 300.375162L266.565125 819.058374c-11.338233 12.190647-11.035334 32.285311 0.638543 44.850487l80.46666 86.564541c11.680017 12.583596 30.356378 12.893658 41.662889 0.716314l377.434212-421.426145c11.332093-12.183484 11.041474-32.266891-0.657986-44.844348l-80.46666-86.564541c-1.772366-1.910513-3.706415-3.533476-5.750981-4.877077L373.270379 71.774697C361.493148 60.273758 343.054193 61.470003 332.091514 74.487481z"></path>
</svg>

            </i>
          </a>
      </nav>
    </footer>
  </article>

  
  

  
  

  

  
  

  

  

  <div class="disqus-comment">
  <div class="disqus-button" id="load_disqus" onclick="load_disqus()">
    显示 Disqus 评论
  </div>
  <div id="disqus_thread"></div>
  <script type="text/javascript">
    var disqus_config = function () {
      this.page.url = "https://jianzuoyi.github.io/post/2020-09-14.microbe-vsearch/";
    };
    function load_disqus() {
      
      
      if (window.location.hostname === 'localhost') return;

      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      var disqus_shortname = 'jianzuoyi';
      dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);

      $('#load_disqus').remove();
    };
  </script>
  <noscript>Please enable JavaScript to view the
    <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a>
  </noscript>
  
  </div>

    

  

        </div>
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="icon-links">
  
  
    <a href="mailto:jianzuoyi@qq.com" rel="me noopener" class="iconfont"
      title="email" >
      <svg class="icon" viewBox="0 0 1451 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M664.781909 681.472759 0 97.881301C0 3.997201 71.046997 0 71.046997 0L474.477909 0 961.649408 0 1361.641813 0C1361.641813 0 1432.688811 3.997201 1432.688811 97.881301L771.345323 681.472759C771.345323 681.472759 764.482731 685.154773 753.594283 688.65053L753.594283 688.664858C741.602731 693.493018 729.424896 695.068979 718.077952 694.839748 706.731093 695.068979 694.553173 693.493018 682.561621 688.664858L682.561621 688.65053C671.644501 685.140446 664.781909 681.472759 664.781909 681.472759L664.781909 681.472759ZM718.063616 811.603883C693.779541 811.016482 658.879232 802.205449 619.10784 767.734955 542.989056 701.759633 0 212.052267 0 212.052267L0 942.809523C0 942.809523 0 1024 83.726336 1024L682.532949 1024 753.579947 1024 1348.948139 1024C1432.688811 1024 1432.688811 942.809523 1432.688811 942.809523L1432.688811 212.052267C1432.688811 212.052267 893.138176 701.759633 817.019477 767.734955 777.248 802.205449 742.347691 811.03081 718.063616 811.603883L718.063616 811.603883Z"></path>
</svg>

    </a>
  
    <a href="https://github.com/jianzuoyi" rel="me noopener" class="iconfont"
      title="github"  target="_blank"
      >
      <svg class="icon" style="" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M512 12.672c-282.88 0-512 229.248-512 512 0 226.261333 146.688 418.133333 350.08 485.76 25.6 4.821333 34.986667-11.008 34.986667-24.618667 0-12.16-0.426667-44.373333-0.64-87.04-142.421333 30.890667-172.458667-68.693333-172.458667-68.693333C188.672 770.986667 155.008 755.2 155.008 755.2c-46.378667-31.744 3.584-31.104 3.584-31.104 51.413333 3.584 78.421333 52.736 78.421333 52.736 45.653333 78.293333 119.850667 55.68 149.12 42.581333 4.608-33.109333 17.792-55.68 32.426667-68.48-113.706667-12.8-233.216-56.832-233.216-253.013333 0-55.893333 19.84-101.546667 52.693333-137.386667-5.76-12.928-23.04-64.981333 4.48-135.509333 0 0 42.88-13.738667 140.8 52.48 40.96-11.392 84.48-17.024 128-17.28 43.52 0.256 87.04 5.888 128 17.28 97.28-66.218667 140.16-52.48 140.16-52.48 27.52 70.528 10.24 122.581333 5.12 135.509333 32.64 35.84 52.48 81.493333 52.48 137.386667 0 196.693333-119.68 240-233.6 252.586667 17.92 15.36 34.56 46.762667 34.56 94.72 0 68.522667-0.64 123.562667-0.64 140.202666 0 13.44 8.96 29.44 35.2 24.32C877.44 942.592 1024 750.592 1024 524.672c0-282.752-229.248-512-512-512"></path>
</svg>

    </a>
  
    <a href="https://www.zhihu.com/people/jianzuoyi" rel="me noopener" class="iconfont"
      title="zhihu"  target="_blank"
      >
      <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="36" height="36">
  <path d="M351.791182 562.469462l192.945407 0c0-45.367257-21.3871-71.939449-21.3871-71.939449L355.897709 490.530013c3.977591-82.182744 7.541767-187.659007 8.816806-226.835262l159.282726 0c0 0-0.86367-67.402109-18.578124-67.402109s-279.979646 0-279.979646 0 16.850783-88.141456 39.318494-127.053698c0 0-83.60514-4.510734-112.121614 106.962104S81.344656 355.077018 76.80834 367.390461c-4.536316 12.313443 24.62791 5.832845 36.941354 0 12.313443-5.832845 68.050885-25.924439 84.252893-103.69571l86.570681 0c1.165546 49.28652 4.596691 200.335724 3.515057 226.835262L109.86113 490.530013c-25.275663 18.147312-33.701566 71.939449-33.701566 71.939449L279.868105 562.469462c-8.497535 56.255235-23.417339 128.763642-44.275389 167.210279-33.05279 60.921511-50.55235 116.65793-169.802314 212.576513 0 0-19.442818 14.257725 40.829917 9.073656 60.273758-5.185093 117.305683-20.739347 156.840094-99.807147 20.553105-41.107233 41.805128-93.250824 58.386782-146.138358l-0.055259 0.185218 167.855986 193.263655c0 0 22.035876-51.847855 5.832845-108.880803L371.045711 650.610918l-42.1244 31.157627-0.045025 0.151449c11.69946-41.020252 20.11206-81.5749 22.726607-116.858498C351.665315 564.212152 351.72876 563.345412 351.791182 562.469462z"></path>
  <path d="M584.918753 182.033893l0 668.840094 70.318532 0 28.807093 80.512708 121.875768-80.512708 153.600307 0L959.520453 182.033893 584.918753 182.033893zM887.150192 778.934538l-79.837326 0-99.578949 65.782216-23.537066-65.782216-24.855084 0L659.341766 256.673847l227.807403 0L887.149169 778.934538z"></path>
</svg>

    </a>


<a href="https://jianzuoyi.github.io/index.xml" rel="noopener alternate" type="application/rss&#43;xml"
    class="iconfont" title="rss" target="_blank">
    <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="30" height="30">
  <path d="M819.157333 1024C819.157333 574.592 449.408 204.8 0 204.8V0c561.706667 0 1024 462.293333 1024 1024h-204.842667zM140.416 743.04a140.8 140.8 0 0 1 140.501333 140.586667A140.928 140.928 0 0 1 140.074667 1024C62.72 1024 0 961.109333 0 883.626667s62.933333-140.544 140.416-140.586667zM678.784 1024h-199.04c0-263.210667-216.533333-479.786667-479.744-479.786667V345.173333c372.352 0 678.784 306.517333 678.784 678.826667z"></path>
</svg>

  </a>
   
</div>

<div class="copyright">
  <span class="power-by">
    Powered by <a class="hexo-link" href="https://gohugo.io">Hugo</a>
  </span>
  <span class="division">|</span>
  <span class="theme-info">
    Theme - <a class="theme-link" href="https://github.com/xianmin/hugo-theme-jane">Jane</a>
  </span>

  <span class="copyright-year">
    &copy;
    2020
    <span class="heart">
      
      <i class="iconfont">
        <svg class="icon" viewBox="0 0 1025 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="14" height="14">
  <path d="M1000.1 247.9c-15.5-37.3-37.6-70.6-65.7-98.9-54.4-54.8-125.8-85-201-85-85.7 0-166 39-221.4 107.4C456.6 103 376.3 64 290.6 64c-75.1 0-146.5 30.4-201.1 85.6-28.2 28.5-50.4 61.9-65.8 99.3-16 38.8-24 79.9-23.6 122.2 0.7 91.7 40.1 177.2 108.1 234.8 3.1 2.6 6 5.1 8.9 7.8 14.9 13.4 58 52.8 112.6 102.7 93.5 85.5 209.9 191.9 257.5 234.2 7 6.1 15.8 9.5 24.9 9.5 9.2 0 18.1-3.4 24.9-9.5 34.5-30.7 105.8-95.9 181.4-165 74.2-67.8 150.9-138 195.8-178.2 69.5-57.9 109.6-144.4 109.9-237.3 0.1-42.5-8-83.6-24-122.2z"
   fill="#8a8a8a"></path>
</svg>

      </i>
    </span><span class="author">
        简佐义
        
      </span></span>

  
  
    <span id="busuanzi_container">
      访客数/访问量：<span id="busuanzi_value_site_uv"></span>/<span id="busuanzi_value_site_pv"></span>
    </span>
  

  
</div>

    </footer>

    <div class="back-to-top" id="back-to-top">
      <i class="iconfont">
        
        <svg class="icon" viewBox="0 0 1024 1024" version="1.1"
  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
  width="35" height="35">
  <path d="M510.866688 227.694839 95.449397 629.218702l235.761562 0-2.057869 328.796468 362.40389 0L691.55698 628.188232l241.942331-3.089361L510.866688 227.694839zM63.840492 63.962777l894.052392 0 0 131.813095L63.840492 195.775872 63.840492 63.962777 63.840492 63.962777zM63.840492 63.962777"></path>
</svg>

      </i>
    </div>
  </div>
  
<script type="text/javascript" src="/lib/jquery/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="/lib/slideout/slideout-1.0.1.min.js"></script>




<script type="text/javascript" src="/js/main.638251f4230630f0335d8c6748e53a96f94b72670920b60c09a56fdc8bece214.js" integrity="sha256-Y4JR9CMGMPAzXYxnSOU6lvlLcmcJILYMCaVv3Ivs4hQ=" crossorigin="anonymous"></script>












  
    <script type="text/javascript" src="/js/load-photoswipe.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe.min.js"></script>
    <script type="text/javascript" src="/lib/photoswipe/photoswipe-ui-default.min.js"></script>
  




  <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>












</body>
</html>
